AMW_business/src/test/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/CopyResourceTest.java

```diff
@@ -84,11 +84,26 @@ public void shouldNotAllowCopyFromResourceOfDifferentType() throws Exception {
 
         // when // then
         copyResource.doCopyResource(targetResourceId, originResourceId, ForeignableOwner.getSystemOwner());
+    }
+
+    @Test(expected = NotAuthorizedException.class)
+    public void shouldThrowNotAuthorizedExceptionWhenPermissionIsDenied() throws ForeignableOwnerViolationException, AMWException {
+        // given
+        ResourceEntity originResourceEntity = mock(ResourceEntity.class);
+        ResourceEntity targetResourceEntity = mock(ResourceEntity.class);
+        ResourceTypeEntity resourceType = mock(ResourceTypeEntity.class);
+        when(targetResourceEntity.getResourceType()).thenReturn(resourceType);
+        when(originResourceEntity.getResourceType()).thenReturn(resourceType);
 
+        // when
+        copyResource.doCopyResource(targetResourceEntity, originResourceEntity, ForeignableOwner.AMW);
+
+        // then
+        verify(copyResourceDomainService, never()).copyFromOriginToTargetResource(originResourceEntity, targetResourceEntity, ForeignableOwner.AMW);
```

Comment:
I think this code is never executes as copyResource throws an exception. You could use assertThrows here:
```
assertThrows(NotAuthorizedException.class, () -> {
        copyResource.doCopyResource(targetResourceEntity, originResourceEntity, ForeignableOwner.AMW);
    });
```

Reply:
ah nice - thank you!